File: Python/myhexdump.py
# Author: tyros77
# Simple hex dump program in python completed as a school project

#!/usr/bin/python3

import sys, os

fh = open(sys.argv[1], 'rb')

def hexdump(start):
    offset = start
    for count in range(6):
        contents = fh.read(16)
        hexa = contents.hex()
        decim = list(contents)
        print(f"[{format(offset, 'x').rjust(8, '0')}]", end=" ")
        if len(hexa) != 32:
            diff = 32-len(hexa)
            buff = '00'*int((diff/2))
            hexa = hexa + buff
        for x in range(0,len(hexa),2):
            print(hexa[x:(x+2)], sep='', end=' ')
        print(' ', end='')
        if len(decim) != 16:
            diff = 16-len(decim)
            for x in range(diff):
                decim.append(0)
        for item in decim:
            if 33 < item < 126:
                print(chr(item), sep='', end='') 
            else:
                print('.', sep='', end='')
        offset += 16
        print('', end='\n')
    print('Total length ', offset)
    return offset
    
newoff = hexdump(0)
size = os.path.getsize(sys.argv[1])
size1 = int(size/16)
for x in range(size1):
    user = input('Continue = c, Quit = q: ')
    if user == 'c':
        newoff = hexdump(int(newoff))
    elif user == 'q':
        sys.exit()
    else:
        print('Invalid syntax')
        sys.exit()

File: C/myhexdump.c
// Author: tyros77
// Simple hex dump completed as a school project

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#define SIZE 16

int main(int argc, char **filename)
{
    FILE *openFile;
    char entry[20];
    char temp[17];
    int count = 0, row, index = 0;
 
    openFile = fopen(filename[1], "r");
    if(openFile == NULL)
    {
        fprintf(stderr, "Could not open file %s.\n", filename[1]);
        exit(1);
    }
    printf("Offset\t Hexadecimal data format\t\t\t\t Character Format\n");
    while(fread(temp, 1, 16, openFile) != 0) 
    {
        printf("%05X\t ", count); //prints the offset for each loop through

        for(row = 0; row < SIZE; row++)
        {
            if(temp[row] < 33)
            {
            temp[row] = '.';
            }
        }
        for(row = 0; row < SIZE; row++)
        {
            printf("%X ", temp[row]);
        }
        printf("\t %s\n", temp);
      
        count = count +16; 
        
        for(row = 0; row < SIZE; row++)  
        {                               
            temp[row] = 0;              
        }
    }

    fclose(openFile);

    return 0;
}
